### PR TITLE
Bugfix/LS24004086/MOVEA from Array field to Standalone

### PR DESCRIPTION
**Pull request by davidepalladino-apuliasoft**
## Description
This work adds the capability to use `MOVEA` from an array declared as DS' field to a Standalone definition. For example:

```
     D AUTOAP        E DS                  EXTNAME(AUTOAP0F)
     D  AU1                   61     80         DIM(10)
     D £AUATI          S             20

     ...

     C                   MOVEA     AU1           £AUATI
```

Also, provides to fix the "overlay" of a DS field, imported by using `EXTNAME`, and another declared with offset. Under the hood, by considering previous example, we have a DS built with fields provided from `AUTOAP0F` and a new field (`AU1`) declared with offset. This offset corresponds with some field provided from `AUTOAP0F`. So, the value for `AU1` might be the value for another field.

### Technical notes
For the first cause, related to `MOVEA`, I fixed the problem by adding to `ProjectedArrayValue` class the implementation for `elementSize`, `takeAll`, `takeFirst`, `takeLast`.

For the second cause, "overlay" of two field by using `EXTNAME`:
1. I improved the construction of `knowDataDefinitions` from files by ignoring data definitions declared without `EXTNAME`;
2. I provided to `RpgParser.Parm_fixedContext.toFieldInfo` the fields declared in a file imported with `EXTNAME`.

Related to #LS24004086

## Checklist:
- [x] If this feature involves RPGLE fixes or improvements, they are well-described in the summary.
- [x] There are tests for this feature.
- [x] RPGLE code used for tests is easily understandable and includes comments that clarify the purpose of this feature.
- [x] The code follows Kotlin conventions (run `./gradlew ktlintCheck`).
- [x] The code passes all tests (run `./gradlew check`).
- [ ] Relevant documentation is included in the `docs` directory.

### CODE REVIEW

File: rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00117.rpgle
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00117.rpgle:1]: What is the goal of this test and in which test unit is it?

**Review by lanarimarco — CHANGES_REQUESTED** [review]

File: rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00116.rpgle [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    O *  Issue executing MoveAStmt at line 32. An operation
+    O *   is not implemented: takeFirst not yet implemented
+    O *   for ProjectedArrayValue.
+     V* ==============================================================
+     D MSG             S             30
+     D COUNT           S              2  0 INZ(1)
+     D AU1_SIZE        S              2  0 INZ(10)
+
+     D AUTOAP        E DS                  EXTNAME(AUTOAP0F)
+     D  AU1                   61     80         DIM(10)
```
[the remark this answers is not in the document]
Comment (davidepalladino-apuliasoft) [reply rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00116.rpgle:19]: In my opinion, I suggest to leave the size to `10` which is an overlay (thanks to start/end offset, too) of `AA£Oxx`, where `xx` is a number from `01` to `10`. Also, this test is like the real scenario. Changing the size implicates a little rewrite of RPG code and a re-execution on `AS400` to ensure the results. Are you sure?

File: rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00117.rpgle
Comment (davidepalladino-apuliasoft) [reply rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00117.rpgle:1]: I forgot to add header with description. I'm going to add.

File: rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00118.rpgle
[the remark this answers is not in the document]
Comment (davidepalladino-apuliasoft) [reply rpgJavaInterpreter-core/src/test/resources/smeup/MUDRNRAPU00118.rpgle:1]: I forgot to add header with description. I'm going to add.

**Review by lanarimarco — CHANGES_REQUESTED** [review]
Maybe in the merge something went wrong.
These tests:
https://github.com/smeup/jariko/commit/08b3413dcc7e159837597eebe628d7235bc26d37
are disappeared.

**Pull request merged**